### PR TITLE
Add msgstake_flow.md

### PR DESCRIPTION
**Pull request by msmania**

### CODE REVIEW

File: doc/specs/msgstake_flow.md [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+flowchart LR
+  A(baseApp/runTx) --> B(x/auth/ValidateTransaction)
+	B --> C{After the upgrade?}
+	C -- yes --> D{Signed by any of:\nthe new output address\nthe operator address\nthe current output address}
+  D -- yes --> F([ok])
+  D -- no --> G([sdk:4])
+	C -- no --> E{Signed by any of:\nthe new output address\nthe operator address}
+  E -- yes --> F
+  E -- no --> G
+  A --> H(baseApp/runMsg)
```
Comment (Olshansk) [inline doc/specs/msgstake_flow.md]: `runMsg` only happens after we `validateTransaction`. 

I think:
1. Splitting it into two flowcharts would make it easier
2. Making the flowcharts `TD` instead o `LR`
3. In `(F[ok])`, making it `see runMsg`

File: doc/specs/msgstake_flow.md [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+A staked node's output address can be changed only when a transaction message
+`MsgStake` is signed by the owner of the node's current output address.  If a
+transaction is signed by the operator address, or the new output address
+specified in the transaction, it's rejected by the blockchain.  The following
+chart illustrates how a `MsgStake` is handled by the key functions.
+
+```mermaid
+flowchart LR
+  A(baseApp/runTx) --> B(x/auth/ValidateTransaction)
+	B --> C{After the upgrade?}
```
Comment (Olshansk) [inline doc/specs/msgstake_flow.md]: In a second line, specify which upgrade

File: doc/specs/msgstake_flow.md (hunk 1 above)
Comment (Olshansk) [inline doc/specs/msgstake_flow.md]: Don't be afraid to split into multiple smaller diagrams. It makes it easier to navigate, share and read.

For example, just scroll (no reason to dive into the details) of https://github.com/pokt-network/pocket/wiki/Utility-Relay-Protocol

File: doc/specs/msgstake_flow.md [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+	C -- yes --> D{Signed by any of:\nthe new output address\nthe operator address\nthe current output address}
+  D -- yes --> F([ok])
+  D -- no --> G([sdk:4])
+	C -- no --> E{Signed by any of:\nthe new output address\nthe operator address}
+  E -- yes --> F
+  E -- no --> G
+  A --> H(baseApp/runMsg)
+  H --> I(x/nodes/handleStake)
+  I --> J(x/nodes/keeper/ValidateValidatorStaking)
+  J --> K{Aafter the upgrade?\nChanging the output address?\nSigned by the current output address?}
```
Comment (Olshansk) [inline doc/specs/msgstake_flow.md]:
```suggestion
  J --> K{After the upgrade?\nChanging the output address?\nSigned by the current output address?}
```

File: doc/specs/msgstake_flow.md [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+  K -- All yes --> L([ok])
+  K -- No --> M(x/nodes/keeper/ValidateValidatorMsgSigner\nwith the new state)
```
Comment (Olshansk) [inline doc/specs/msgstake_flow.md]:
```suggestion
  K -- Any no --> M(x/nodes/keeper/ValidateValidatorMsgSigner\nwith the new state)
```

File: doc/specs/msgstake_flow.md [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+  R -- No --> P
+  J --> S(x/nodes/keeper/ValidateEditStake)
+	S -- We're changing the output address --> T{After the upgrade?}
+	T -- yes --> U{Signed by the current output address?}
+	U -- yes --> V([ok])
+	U -- no --> W([pos:127])
+	T -- no --> X([pos:124])
+  H --> Y(x/nodes/keeper/StakeValidator)
+  Y --> Z(x/nodes/keeper/EditStakeValidator)
+  Z --> AA([Happy ending!])
```
Comment (Olshansk) [inline doc/specs/msgstake_flow.md]: giphy

File: doc/specs/msgstake_flow.md (hunk 3 above)
Comment (Olshansk) [inline doc/specs/msgstake_flow.md]: Screenshot 2023-04-10 at 4 07 38 PM

It's kind of hard to understand the flow here and what's expected. 

Again, I think multiple small TD diagrams will help clarify it

**Review by Olshansk — CHANGES_REQUESTED** [review]

File: doc/specs/msgstake_flow.md (hunk 3 above)
Comment (msmania) [reply doc/specs/msgstake_flow.md]: Okay, split into four charts and added simple descriptions in each section.

**Review by Olshansk — APPROVED** [review]
THIS IS SO GOOD!!! 🙏 🙏 🙏 🙏

**Comment by msmania** [comment]
I feel this is an excessive effort, but yeah, it looks much better and hopefully this helps someone.  Thank you for your thorough review as always!

**Pull request merged**